runtime/lua/vim/filetype/detect.lua

```diff
@@ -0,0 +1,452 @@
+local M = {}
+
+---@private
+local function count_lines(bufnr)
+  return vim.api.nvim_buf_line_count(bufnr)
+end
+
+---@private
+local function get_lines(bufnr, start_lnum, end_lnum, opts)
+  if not end_lnum then
+    -- Return a single line as a string
+    return vim.api.nvim_buf_get_lines(bufnr, start_lnum - 1, start_lnum, false)[1]
+  end
+
+  local lines = vim.api.nvim_buf_get_lines(bufnr, start_lnum - 1, end_lnum, false)
+  opts = opts or {}
+  return opts.concat and (table.concat(lines) or "") or lines
+end
+
+function M.asm(path, bufnr)
+
+end
+
+function M.asm_syntax(path, bufnr)
+
+end
+
+function M.bas(path, bufnr)
+
+end
+
+function M.bindzone(path, bufnr)
+
+end
+
+function M.btm(bufnr)
+  if vim.g.dosbatch_syntax_for_btm and vim.g.dosbatch_syntax_for_btm ~= 0 then
+    vim.bo[bufnr].filetype = "dosbatch"
+  else
+    vim.bo[bufnr].filetype = "btm"
+  end
+end
+
+function M.cfg(path, bufnr)
+  if vim.g.filetype_cfg then
+    vim.bo[bufnr].filetype = vim.g.filetype_cfg
+  elseif M.is_rapid(path, bufnr) then
```

Comment:
`IsRapid` is a "local" function in `dist#ft` so we can make it the same here (whenever you get to it).